### ISSUE TITLE
Model-Exchange FMU 2.0 Export fails for Modelica Buildings Library model

### ISSUE THREAD

**Issue by dhblum** [OpenModelica/OpenModelica#8805]
### Description
Model-Exchange FMU 2.0 Export fails for Modelica Buildings Library model ``Buildings.ThermalZones.Detailed.Validation.BESTEST.Cases6xx.Case600FF``.

### Steps to reproduce
- Open OMEdit
- Load MSL
- Load Modelica Buildings Library
- Open Model ``Buildings.ThermalZones.Detailed.Validation.BESTEST.Cases6xx.Case600FF``
- In Tools > Options > FMI, set Type=Model Exchange.  Keep version = 2.0.
- File > Export > FMU

### Expected behavior
An FMU is generated.

### Additional files
Attaching the log output from "All" tab in Message Browser of OMEdit, which hows error messages.
[exportFMUlog.txt](https://github.com/OpenModelica/OpenModelica/files/8419540/exportFMUlog.txt)

### Version and OS
 - OpenModelica Version 1.19.0~dev-748-g0f91fa0 (updated nightly build install on 4/5/2022)
 - Modelica Standard Library 4.0.0+maint.om
 - Modelica Buildings Library commit 31bdfcee3332f47ef34820a9d9eef8215ecaba3f from master branch
 - OS: Ubuntu 18.04, 64 bit

### Additional context
The model in question compiles and simulates OK in OMEdit.

**Comment by lochel** [comment]
There is some problem with the generated code:
```
/usr/bin/ld: Buildings_ThermalZones_Detailed_Validation_BESTEST_Cases6xx_Case600FF_08bnd.o: in function `Buildings_ThermalZones_Detailed_Validation_BESTEST_Cases6xx_Case600FF_updateBoundParameters':
Buildings_ThermalZones_Detailed_Validation_BESTEST_Cases6xx_Case600FF_08bnd.c:(.text+0x19eaf): undefined reference to `Buildings_ThermalZones_Detailed_Validation_BESTEST_Cases6xx_Case600FF_updateBoundParameters_0'
/usr/bin/ld: Buildings_ThermalZones_Detailed_Validation_BESTEST_Cases6xx_Case600FF_08bnd.c:(.text+0x19ebd): undefined reference to `Buildings_ThermalZones_Detailed_Validation_BESTEST_Cases6xx_Case600FF_updateBoundParameters_1'
/usr/bin/ld: Buildings_ThermalZones_Detailed_Validation_BESTEST_Cases6xx_Case600FF_08bnd.c:(.text+0x19ecb): undefined reference to `Buildings_ThermalZones_Detailed_Validation_BESTEST_Cases6xx_Case600FF_updateBoundParameters_2'
/usr/bin/ld: Buildings_ThermalZones_Detailed_Validation_BESTEST_Cases6xx_Case600FF_08bnd.c:(.text+0x19ed9): undefined reference to `Buildings_ThermalZones_Detailed_Validation_BESTEST_Cases6xx_Case600FF_updateBoundParameters_3'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by casella** [comment]
@mahge, apparently this regression creeped in some time after 1.18.1, see [regression report](https://libraries.openmodelica.org/branches/overview-combined-fmi.html), e.g. for the Buildings_latest Library
![immagine](https://user-images.githubusercontent.com/4233148/162930941-8cf51295-f704-4a79-9c42-a348d9dd5cf6.png)

The C runtime has no such problem, see [regression report](https://libraries.openmodelica.org/branches/overview.html)

![immagine](https://user-images.githubusercontent.com/4233148/162931133-abe09541-85bf-44f2-a40a-685207eddece.png)

so it is some library-related issue with FMI only. This is quite a bad regression, could please have a look? Thanks!

**Comment by casella** [comment]
@adrpo thinks that `Buildings_ThermalZones_Detailed_Validation_BESTEST_Cases6xx_Case600FF_updateBoundParameters` is split into multiple functions, which are handled correctly for the C runtime, but not for FMI.

Please check!

**Comment by casella** [comment]
Thanks @mahge!

**Comment by dhblum** [comment]
Thank you @mahge and @casella for addressing so quickly!

**Comment by dhblum** [comment]
I've updated my nightly build today to OpenModelica 1.19.0~dev-765-g5b322f1 and I'm still getting the same error when exporting from OMEdit.  Would this fix have made it into that build yet?

**Comment by casella** [comment]
> I've updated my nightly build today to OpenModelica 1.19.0~dev-765-g5b322f1 and I'm still getting the same error when exporting the FMU using OMEdit. Would this fix have made it into that build yet?

No, it's not, as you can see from the [commit history](https://github.com/OpenModelica/OpenModelica/commits/master) ecf58a9 came much later than 5b322f1. You can try the latest nightly build, or the 1.19.0-beta.1 release.

**Comment by dhblum** [comment]
@casella Ohh, now I see how the nightly build suffix "g5b322f1" is linked to the latest commit number.  Sounds good, thank you for the clarification!

**Comment by dhblum** [comment]
Indeed, this is fixed, tested with OpenModelica 1.20.0~dev.
